### PR TITLE
Sync cluster CA to `ConfigMap` and use it in `shoots/{admin,viewer}kubeconfig` subresource

### LINKED ISSUE #9091: Store `Shoot` cluster CA bundle as `ConfigMap`
**How to categorize this issue?**

/area usability
/area security
/kind enhancement

**What would you like to be added**:
The `Shoot` cluster CA bundle should be stored as `ConfigMap` so that it can be read by users with `viewer` `Project` role. The CA bundle is not sensitive data, hence it doesn't need to be stored as a `Secret`.

**Why is this needed**:
Gardener stores a `Secret` with the name `<shoot-name>.ca-cluster` in the project namespace in the garden cluster. This `Secret` contains the CA bundle of the cluster. The `dashboard` and `gardenctl` generate a `gardenlogin` kubeconfig by reading the CA data from the `Secret` and mapping it to the `certificate-authority-data` field.

Gardenlogin will soon also support the `shoot/viewerkubeconfig` subresource ([#78](https://github.com/gardener/gardenlogin/pull/78)) which enables users with `viewer` role to have read-only access to a cluster. However, users with `viewer` role using `gardenctl` or the `dashboard` do not have the permission to read this `Secret`.

A potential workaround for generating the `gardenlogin` kubeconfig could involve calling the subresource `shoot/adminkubeconfig` or `shoot/viewerkubeconfig` solely for the `certificate-authority-data` field and discarding the rest. However, this approach might lead to unnecessary audit events.

### PR DESCRIPTION
**Pull request by petersutter**
**How to categorize this PR?**

/area usability
/area security
/kind enhancement

**What this PR does / why we need it**:
With this PR the cluster CA is synced to a `<shoot-name>.ca-cluster` `ConfigMap` in the project namespace of the garden cluster. The `shoots/{admin,viewer}kubeconfig` subresource retrieves it from there instead of using the `Secret` resource.
It falls back to the `Secret` if the `ConfigMap` has not been synced yet.

**Which issue(s) this PR fixes**:
Fixes #9091

**Special notes for your reviewer**:

**Release note**:

```breaking developer
The util function `pkg/utils/gardener.ComputeShootProjectSecretName` has been renamed to `pkg/utils/gardener.ComputeShootProjectResourceName`.
```

```noteworthy user
The shoot cluster CA bundle is now stored in a `ConfigMap` in the project namespace of the garden cluster, in addition to storing it in a `Secret`. This `ConfigMap` shares the same name as the pre-existing `Secret`, which is `<shoot-name>.ca-cluster`. The `Secret` will be removed in a future Gardener release. Therefore, if your tooling relies on this `Secret`, you should update it to fetch the `ConfigMap` instead.
```

### CODE REVIEW

File: pkg/operation/botanist/secrets.go [hunk 1]
```diff
@@ -216,12 +216,25 @@ func (b *Botanist) generateCertificateAuthorities(ctx context.Context) error {
...
+		ctx,
+		gardenerutils.ShootProjectConfigMapSuffixCACluster,
+		map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleCACluster},
+		nil,
+		map[string]string{secretsutils.DataKeyCertificateCA: string(caBundle)},
+	); err != nil {
+		return err
+	}
+
+	// TODO(petersutter): Remove this code and cleanup Secret in a future release. The caBundle is now being stored in a <shootname>.ca-cluster ConfigMap.
```
Comment (rfranzke) [inline pkg/operation/botanist/secrets.go]:
```suggestion
	// TODO(petersutter): Remove this code and cleanup Secret after v1.96 has been released. The caBundle is now being stored in a <shootname>.ca-cluster ConfigMap.
```

Maybe you can make yourself a reminder for CW23 :)

File: pkg/registry/core/shoot/storage/kubeconfig.go [hunk 2]
```diff
@@ -120,14 +121,23 @@ func (r *KubeconfigREST) Create(ctx context.Context, name string, obj runtime.Ob
 		return nil, apierrors.NewInternalError(fmt.Errorf("could not load client CA certificate from secret: %w", err))
 	}
 
-	caClusterSecret, err := r.secretLister.Secrets(shoot.Namespace).Get(gardenerutils.ComputeShootProjectResourceName(shoot.Name, gardenerutils.ShootProjectSecretSuffixCACluster))
-	if err != nil {
-		return nil, apierrors.NewInternalError(fmt.Errorf("could not get cluster CA secret: %w", err))
+	var clusterCABundle []byte
+	caClusterConfigMap, err := r.configMapLister.ConfigMaps(shoot.Namespace).Get(gardenerutils.ComputeShootProjectResourceName(shoot.Name, gardenerutils.ShootProjectConfigMapSuffixCACluster))
+	// TODO(petersutter): remove this fallback of reading the <shoot-name>.ca-cluster Secret in a future release
```
Comment (rfranzke) [inline pkg/registry/core/shoot/storage/kubeconfig.go]: v1.96

File: pkg/registry/core/shoot/storage/kubeconfig.go [hunk 3]
```diff
@@ -44,6 +44,7 @@ import (
 type KubeconfigREST struct {
 	secretLister         kubecorev1listers.SecretLister
 	internalSecretLister gardencorelisters.InternalSecretLister
+	configMapLister      kubecorev1listers.ConfigMapLister
```
Comment (rfranzke) [inline pkg/registry/core/shoot/storage/kubeconfig.go:48]: Also add a TODO statement to remove this struct attribute

**Review by rfranzke — CHANGES_REQUESTED** [review]
Impressive PR @petersutter, nice! Thanks for the great commit structure!

Could you please also adapt the occurrences in `test/e2e` ([example](https://github.com/gardener/gardener/blob/aa2f8c0cd392420f3a74f33d5098478a66fad80d/test/e2e/gardener/shoot/internal/rotation/certificate_authorities.go#L93))?

File: pkg/operation/botanist/secrets.go (hunk 1 above)
Comment (petersutter) [reply pkg/operation/botanist/secrets.go]: ✅  fixed and I created myself a reminder :)

File: pkg/registry/core/shoot/storage/kubeconfig.go (hunk 2 above)
Comment (petersutter) [reply pkg/registry/core/shoot/storage/kubeconfig.go]: ✅

File: pkg/registry/core/shoot/storage/kubeconfig.go (hunk 3 above)
Comment (petersutter) [reply pkg/registry/core/shoot/storage/kubeconfig.go:48]: ✅

**Comment by rfranzke** [comment]
Thanks, let's get this merged after you rebased and the tests are fixed.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**